extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -0,0 +1,378 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assertHttpsUrl} from '../../../src/url';
+import {assert} from '../../../src/asserts';
+import {cidFor} from '../../../src/cid';
+import {getService} from '../../../src/service';
+import {isExperimentOn} from '../../../src/experiments';
+import {log} from '../../../src/log';
+import {onDocumentReady} from '../../../src/document-state';
+import {viewerFor} from '../../../src/viewer';
+import {xhrFor} from '../../../src/xhr';
+
+
+/** @const */
+const EXPERIMENT = 'amp-user-notification';
+
+/**
+ * @export
+ * @typedef {{
+ *   elementId: string,
+ *   ampUserId: string
+ * }}
+ */
+let PostRequestMetadataDef;
+
+/**
+ * @export
+ * @typedef {{
+ *   showNotification: boolean
+ * }}
+ */
+let PostResponseMetadataDef;
+
+/**
+ * @typedef {{
+ *   promise: !Promise,
+ *   resolve: function(*)
+ * }}
+ */
+let UserNotificationDeferDef;
+
+
+/**
+ * Defines underlying API for Notification components.
+ * @interface
+ */
+class NotificationInterface {
+
+  /**
+   * Promise that is resolved with a boolean on whether this Notification
+   * should be shown or not.
+   * @return {!Promise<boolean>}
+   */
+  shouldShow() {}
+
+  /**
+   * Turns the Notification Component visible.
+   * @return {!Promise}
+   */
+  show() {}
+}
+
+/**
+ * Component class that handles a simple notification.
+ * @implements {NotificationInterface}
+ */
+export class AmpUserNotification extends AMP.BaseElement {
+
+  /**
+   * @private
+   */
+  isExperimentOn_() {
+    return isExperimentOn(this.win_, EXPERIMENT);
```

Comment:
Is this guaranteed to be always called after `buildCallback` where the `this.win_` is initialized? If not, please replace with `this.getWin()`